pkg/controller/compute-machineset/compute_machineset_controller.go

```diff
@@ -0,0 +1,519 @@
+	}
+
+	// update Status field
+	/***
+	original := ms
+	newms = ms.DeepCopy()
+	***/
+
+	if manageMachinesErr != nil {
+		return manageMachinesErr
+	}
+
+
+	return nil
+}
+
+func (c *Controller) manageMachines(machines []*clusteroperator.Machine, ms *clusteroperator.MachineSet) error {
+	machineSetKey, err := controller.KeyFunc(ms)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for MachineSet %#v: %v", ms, err))
+		return nil
+	}
+
+	machinePrefix := ms.Name + "-"
+	//c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine prefix: %v", machinePrefix)
+
+	machineSetMachines := []*clusteroperator.Machine{}
+	//machinesToCreate := []*clusteroperator.Machine{}
+	machinesToDelete := []string{}
+	//machinesToUpdate := []*clusteroperator.Machine{}
+
+	deleteCount := int(0)
+	createCount := int(0)
+
+
+	// caculate how many to create/delete
+	if len(machines) > ms.Spec.Size {
+		deleteCount = len(machines) - ms.Spec.Size
+	}
+	if len(machines) < ms.Spec.Size {
+		createCount = ms.Spec.Size - len(machines)
+	}
+
+	for _, machine := range machines {
+		/*
+		if currentMachineNeedsUpdate(machine, ms) {
+			c.logger.WithField("machineset", ms.Name).Errorf("JDIAZ child machine needs update %v", machine.Name)
+			machinesToDelete = append(machinesToDelete, machine)
+			createCount++
+			deleteCount++
+			continue
+		}
+		*/
+
+		machineSetMachines = append(machineSetMachines, machine)
+
+	}
+
+	for i := 0 ; i < deleteCount; i++ {
+		machinesToDelete = append(machinesToDelete, getMachineKey(machineSetMachines[i]))
+	}
+	c.logger.WithField("machineset", ms.Name).Infof("JDIAZ creating %v, deleting %+v", createCount, machinesToDelete)
+	c.expectations.SetExpectations(machineSetKey, createCount, machinesToDelete)
+
+	// Sync machines
+	machineConfig := buildNewMachine(ms, machinePrefix)
+	c.logger.WithField("machineset", ms.Name).Debugf("JDIAZ machine template: %+v", machineConfig)
+
+	for i := 0 ; i < createCount ; i++ {
+		_, err := c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Create(machineConfig)
+		if err != nil {
+			// FIXME add to error list?
+			c.logger.WithField("machineset", ms.Name).Error("JDIAZ error creating machine...")
+			// update expectations?
+		}
+	}
+
+	c.logger.WithField("machineset", ms.Name).Debugf("JDIAZ delete count :%v", deleteCount)
+	/*
+	for _, m := range machinesToDelete {
+		c.logger.WithField("machineset", ms.Name).Debugf("JDIAZ deleting machine %v", m.Name)
+		c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Delete(m.Name, &metav1.DeleteOptions{})
+		// catch error ^^^
+		deleteCount--
+	}
+	*/
+
+	c.logger.WithField("machineset", ms.Name).Debugf("JDIAZ deleting %v more machines", deleteCount)
+	for i := 0 ; i < deleteCount ; i++ {
+		c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Delete(machineSetMachines[i].Name, &metav1.DeleteOptions{})
+		// catch error ^^^
```

Comment:
Catch error and call `c.expectations.DeletionObserved(machineSetKey, key_for_the_machine)` so that the controller is not waiting to observe the machine that the controller failed to delete.